engine/src/main/java/org/terasology/engine/TerasologyEngine.java

```diff
@@ -131,11 +128,14 @@
     private GameState pendingState;
     private Set<StateChangeSubscriber> stateChangeSubscribers = Sets.newLinkedHashSet();
 
-    private enum EngineState {UNINITIALIZED, INITIALIZED, RUNNING, DISPOSED}
+    private EngineStatus status = StandardGameStatus.UNSTARTED;
+    private final List<EngineStatusSubscriber> statusSubscriberList = new CopyOnWriteArrayList<>();
 
-    private EngineState engineState = EngineState.UNINITIALIZED;
-
-    private final TaskMaster<Task> commonThreadPool = TaskMaster.createFIFOTaskMaster("common", MAX_NUMBER_THREADS);
+    private transient boolean initialised;
+    private transient boolean shutdownRequested;
+    private transient boolean shuttingDown;
+    private transient boolean running;
+    private transient boolean disposed;
```

Comment:
I just went through the lifecycle. 

Let me get this straight: basically the engine is either "running" or "disposed". Within the context of "running" it can be "initialized", a shutdown may have been requested or it is actually in the process of shutting down. Then within the context of "disposed" everything else is false.

If I'm correct, I'm wondering if you could, for starters, remove the "initialized" variable. You use it only to avoid re-initialization but if that's the case could we perhaps prevent the re-execution of run()?

The other thing is, I appreciate the sophistication of having a shutdownRequested and a shuttingDown. But from the code having just "shuttingDown" set by shutdown() and checked by the main loop seem it would work just as well. In this context (eliminated shutdownRequested and initialized), the states "running", "shuttingDown" and "disposed" seem mutually exclusive and could go back into an enum, with the addition of an initial "NotYetStarted" value. This would make the whole lifecycle much easier to define and keep in mind.

Or am I missing something?